simple_slice_rust.rs

```
@@ -0,0 +1,162 @@
+        if let Some(start) = start {
+            if let Ok(slice) = start.extract::<PyRef<SimpleSliceRust>>() {
+                return Ok(Self {
+                    _start: slice._start,
+                    _stop: slice._stop,
+                    _step: slice._step,
+                    _has_start: slice._has_start,
+                    _has_stop: slice._has_stop,
+                    _has_step: slice._has_step,
+                });
+            }
+
+            if let Ok(slice) = start.downcast::<PySlice>() {
+                return Self::new_impl(py,
+                    Some(slice.getattr("start")?),
+                    Some(slice.getattr("stop")?),
+                    Some(slice.getattr("step")?),
+                );
+            }
+        }
+
+        let (mut start, mut stop) = (start, stop);
+
+        if stop.is_none() && start.is_some() {
+            std::mem::swap(&mut start, &mut stop);
+        }
+
+        let py_index = |obj: &PyAny| -> PyResult<isize> {
+            if obj.is_instance_of::<PyBool>() {
+                return Err(PyTypeError::new_err("'bool' object cannot be interpreted as an integer"));
+            }
+            if obj.hasattr("__class__")? && obj.getattr("__class__")?.getattr("__name__")?.extract::<String>()? == "bool_" {
+                return Err(PyTypeError::new_err("'numpy.bool_' object cannot be interpreted as an integer"));
+            }
+            match obj.call_method0("__index__") {
+                Ok(index_obj) => index_obj.extract::<isize>(),
+                Err(_) => Err(PyTypeError::new_err(format!("'{}' object cannot be interpreted as an integer", obj.get_type().name()?)))
+            }
+        };
+
+        let _has_start = start.is_some();
+        let _has_stop = stop.is_some();
+        let _has_step = step.is_some();
+
+        let _start = start.map(py_index).transpose()?.unwrap_or(0);
+        let _stop = stop.map(py_index).transpose()?.unwrap_or(0);
```

This is suppressing all the errors you're raising in the py_index closure